cmd/cluster/server.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/atlassian/gostatsd/pkg/cluster/nodes"
+	"github.com/spf13/pflag"
+)
+
+// Cluster is everything for running a single node in a cluster
+type Cluster struct {
+	RedisAddr      string
+	Namespace      string
+	Target         string
+	UpdateInterval time.Duration
+	ExpiryInterval time.Duration
+}
+
+// newCluster will create a new Cluster with default values.
+func newCluster() *Cluster {
+	local, _ := nodes.LocalAddress("1.1.1.1:1")
+
+	return &Cluster{
+		RedisAddr:      "127.0.0.1:6379",
+		Namespace:      "namespace",
+		Target:         local.String() + ":8125",
+		UpdateInterval: time.Second,
+		ExpiryInterval: 30 * time.Second,
+	}
+}
+
+// AddFlags adds flags for a specific Server to the specified FlagSet.
+func (c *Cluster) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&c.RedisAddr, "redis-addr", c.RedisAddr, "Redis address")
+	fs.StringVar(&c.Namespace, "namespace", c.Namespace, "Namespace")
+	fs.StringVar(&c.Target, "target", c.Target, "Target port")
+	fs.DurationVar(&c.UpdateInterval, "update-interval", c.UpdateInterval, "Cluster update interval")
+	fs.DurationVar(&c.ExpiryInterval, "expiry-interval", c.ExpiryInterval, "Cluster expiry interval")
+}
+
+// Run runs the specified Cluster.
+func (c *Cluster) Run() error {
+	rnt := nodes.NewRedisNodeTracker(c.RedisAddr, c.Namespace, c.Target, c.UpdateInterval, c.ExpiryInterval)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	go rnt.Run(ctx)
+
+	t := time.NewTicker(time.Second)
+
+	for range t.C {
```

Comment:
It's an infinite loop, isn't it?

Reply:
It is.  doing a ```for{ select { } }``` with only one case makes the linter complain though.

Reply:
Why single case? Should also select on `ctx.Done()`.

Reply:
ctx is never closed.  The cluster binary is a test harness only for testing the redis behaviour. Things like graceful shutdown and whatnot aren't going here because it's a lower standard, they'll be handled properly in the main binary.